scout/ci.md

```diff
@@ -0,0 +1,141 @@
+  REGISTRY: docker.io
+  IMAGE_NAME: $\{\{ github.repository \}\}
+  SHA: $\{\{ github.event.pull_request.head.sha || github.event.after \}\}
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+```
+
+This sets up the workflow to run on pull requests and pushes to the `main`
+branch, and sets up environment variables available to all workflow steps. It
+then defines a job called `build` that runs on the latest Ubuntu image and sets
+the permissions available to the job.
+
+Add the following to the YAML file:
+
+```yaml
+  steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+      with:
+        ref: $\{\{ env.SHA \}\}
+
+    - name: Setup Docker buildx
+      uses: docker/setup-buildx-action@v2.5.0
+
+    # Login against a Docker registry except on PR
+    # https://github.com/docker/login-action
+    - name: Log into registry $\{\{ env.REGISTRY \}\}
+      uses: docker/login-action@v2.1.0
+      with:
+        registry: $\{\{ env.REGISTRY \}\}
+        username: $\{\{ secrets.DOCKER_USER \}\}
+        password: $\{\{ secrets.DOCKER_PAT \}\}
+
+    # Extract metadata (tags, labels) for Docker
+    # https://github.com/docker/metadata-action
+    - name: Extract Docker metadata
+      id: meta
+      uses: docker/metadata-action@v4.4.0
+      with:
+        images: $\{\{ env.REGISTRY \}\}/$\{\{ env.IMAGE_NAME \}\}
+        labels: |
+          org.opencontainers.image.revision=$\{\{ env.SHA \}\}
+        tags: |
+          type=edge,branch=$repo.default_branch
+          type=semver,pattern=v\{\{version\}\}
+          type=sha,prefix=,suffix=,format=short
+```
+
+This creates workflow steps to checkout the repository, set up Docker buildx,
+log into the Docker registry, and extract metadata from Git reference and GitHub
+events to use in later steps.
+
+Add the following to the YAML file:
+
+```yaml
+# Build and push Docker image with Buildx (don't push on PR)
+# https://github.com/docker/build-push-action
+- name: Build and push Docker image
+  id: build-and-push
+  uses: docker/build-push-action@v4.0.0
+  with:
+    context: .
+    push: true
+    tags: $\{\{ steps.meta.outputs.tags \}\}
+    labels: $\{\{ steps.meta.outputs.labels \}\}
+    cache-from: type=gha
+    cache-to: type=gha,mode=max
+```
+
+This uses the extracted metadata from the previous step to build and push the
+Docker image to Docker Hub. GitHub Actions skips this step on pull requests and
+only runs when a pull request is merged.
+
+Add the following to the YAML file:
+
+```yaml
+- name: Docker Scout
+  id: docker-scout
+  if: $\{\{ github.event_name == 'pull_request' \}\}
+  uses: docker/scout-action@dd36f5b0295baffa006aa6623371f226cc03e506
+  with:
+    command: compare
+    image: $\{\{ steps.meta.outputs.tags \}\}
+    to: $\{\{ env.REGISTRY \}\}/$\{\{ env.IMAGE_NAME \}\}:edge
+    ignore-unchanged: true
+    only-severities: critical,high
+    token: $\{\{ secrets.DOCKER_PAT \}\}
```

Comment:
@dvdksn @ChrisChinchilla I miss this, why do we have `$\{\{` ? I think we just need a raw statement for this like: https://github.com/docker/docs/blame/6dd952275680cf6f740058f1c0cf0b2e4e25f2ee/build/ci/github-actions/configure-builder.md#L165

Reply:
@crazy-max Without it I got a lot of Liquid render errors. There may be a better way, but this worked for me… Can also switch if you'd rather?

Reply:
Oh. I missed this too. It needs liquid escape: `{% raw %} ... {% endraw %}`

Reply:
can you push an amendment @ChrisChinchilla ?

Reply:
Awwright @dvdksn and @crazy-max done.